pkg/blobstore/metrics_blob_access.go

```diff
@@ -86,12 +89,18 @@ func (ba *metricsBlobAccess) updateDurationSeconds(vec prometheus.ObserverVec, c
 }
 
 func (ba *metricsBlobAccess) Get(ctx context.Context, digest digest.Digest) buffer.Buffer {
+	ctx, span := trace.StartSpan(ctx, ba.name+".Get")
```

Comment:
My suggestion would be to add a separate TracingBlobAccess. There is no need to let this be part of MetricsBlobAccess, right?

The advantage of having a separate TracingBlobAccess is that it allows us to prevent the overhead on systems where tracing is not enabled.

Reply:
There is indeed no need to put it in here, we can add another wrapper for tracing if you prefer.

This code is already near-zero-overhead. The opencensus and opentelemetry libraries handle this explicitly so that applications using them don't need to add extra logic for it. All of the calls here are no-ops if tracing is not enabled. The worst thing we do here is compute ba.name+".Get" and I can smooth that out if you think it's important (I expect the compiler to optimise it away).

There's a general principle here that if any overhead should be removed, it should be removed in the tracing library, rather than expecting applications to try to reproduce it.